streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSessionStore.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.kafka.streams.kstream.internals.SessionKeyBinaryConverter;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.SessionStore;
+import org.apache.kafka.streams.state.StateSerdes;
+
+import java.util.List;
+
+class RocksDBSessionStore<K, AGG> implements SessionStore<K, AGG> {
+
+    private final Serde<K> keySerde;
+    private final Serde<AGG> aggSerde;
+    private final SegmentedBytesStore bytesStore;
+    private StateSerdes<K, AGG> serdes;
+
+
+    RocksDBSessionStore(final SegmentedBytesStore bytesStore,
+                        final Serde<K> keySerde,
+                        final Serde<AGG> aggSerde) {
+        this.keySerde = keySerde;
+        this.bytesStore = bytesStore;
+        this.aggSerde = aggSerde;
+    }
+
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public KeyValueIterator<Windowed<K>, AGG> findSessionsToMerge(final K key, final long earliestSessionEndTime, final long latestSessionStartTime) {
+        final KeyValueIterator<Bytes, byte[]> bytesIterator = bytesStore.fetch(Bytes.wrap(serdes.rawKey(key)), earliestSessionEndTime, latestSessionStartTime);
+        return new SessionStoreIterator(bytesIterator, serdes);
+    }
+
+
+    @Override
+    public void remove(final Windowed<K> key) {
+        bytesStore.remove(SessionKeyBinaryConverter.toBinary(key, serdes.keySerializer()));
+    }
+
+    @Override
+    public void put(final Windowed<K> sessionKey, final AGG aggregate) {
+        bytesStore.put(SessionKeyBinaryConverter.toBinary(sessionKey, serdes.keySerializer()), aggSerde.serializer().serialize(bytesStore.name(), aggregate));
+    }
+
+    @Override
+    public String name() {
+        return bytesStore.name();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void init(final ProcessorContext context, final StateStore root) {
+        this.serdes = new StateSerdes<>(bytesStore.name(),
+                                        keySerde == null ? (Serde<K>) context.keySerde() : keySerde,
+                                        aggSerde == null ? (Serde<AGG>) context.valueSerde() : aggSerde);
+
+        bytesStore.init(context, root);
+    }
+
+    @Override
+    public void flush() {
+        bytesStore.flush();
+    }
+
+    @Override
+    public void close() {
+        bytesStore.close();
+    }
+
+    @Override
+    public boolean persistent() {
+        return true;
+    }
+
+    @Override
+    public boolean isOpen() {
+        return bytesStore.isOpen();
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<K>, AGG> fetch(final K key) {
+        return findSessionsToMerge(key, 0, Long.MAX_VALUE);
+    }
+
+    static class SessionKeySchema implements RocksDBSegmentedBytesStore.KeySchema {
```

Comment:
Is this specific to RocksDB, since it is used in the general CachingSessionState as well?